hyperspy/io.py

```diff
@@ -103,45 +102,52 @@ def load(filenames=None,
         If an axis with this name already
         exists it automatically append '-i', where `i` are integers,
         until it finds a name that is not yet in use.
-    mmap: bool
-        If True and stack is True, then the data is stored
-        in a memory-mapped temporary file.The memory-mapped data is
-        stored on disk, and not directly loaded into memory.
-        Memory mapping is especially useful for accessing small
-        fragments of large files without reading the entire file into
-        memory.
-    mmap_dir : string
-        If mmap_dir is not None, and stack and mmap are True, the memory
-        mapped file will be created in the given directory,
-        otherwise the default directory is used.
-    load_to_memory: bool
-        for HDF5 files, blockfiles and EMD files, if True (default) loads all
-        data to memory. If False, enables only loading the data upon request
-    mmap_mode: {'r', 'r+', 'c'}
-        Used when loading blockfiles to determine which mode to use for when
-        loading as memmap (i.e. when load_to_memory=False)
+    lazy : {None, bool}
+        Open the data lazily - i.e. without actually reading the data from the
+        disk until required. Allows opening arbitrary-sized datasets.
+        If None, default from preferences is used.
 
     print_info: bool
         For SEMPER unf- and EMD (Berkley)-files, if True (default is False)
-        additional information read during loading is printed for a quick overview.
-
+        additional information read during loading is printed for a quick
+        overview.
 
     Returns
     -------
     Signal instance or list of signal instances
+
     Examples
     --------
     Loading a single file providing the signal type:
     >>> d = hs.load('file.dm3', signal_type="EDS_TEM")
 
     Loading multiple files:
+
     >>> d = hs.load('file1.dm3','file2.dm3')
+
     Loading multiple files matching the pattern:
+
     >>> d = hs.load('file*.dm3')
+
+    Loading (potentially larger than the available memory) files lazily and
+    stacking:
+
+    >>> s = hs.load('file*.blo', lazy=True, stack=True)
+
     """
+    deprecated = ['load_to_memory', 'mmap', 'mmap_dir', 'mmap_mode']
+    warn_str = "'{}' argument is deprecated, please use 'lazy' instead"
+    for k in deprecated:
+        if k in kwds:
+            lazy=True
+            warnings.warn(warn_str.format(k), VisibleDeprecationWarning)
+            del kwds[k]
```

Comment:
We cannot simply ignore the keywords as this could break somebody's API. ``load_to_memory`` and ``mmap`` could just set the value of ``lazy``. ``mmap_dir`` could be ignored as this was a poor man solution to the problem and makes no sense now so nobody will miss it. ``mmap_mode`` is trickier: this allows to modify the original file when ``mmap_mode`` is ``r+``, is this available for lazily loaded signals?